File: MauiApp1/Domain/Printing/ReceiptLine.cs
namespace MauiApp1.Domain.Printing;

/// <summary>
/// フォーマット済みレシートの1行を表す値オブジェクト。
/// マーカー（[B][C][R][L][LR]）解析済みの状態を保持する。
/// </summary>
public sealed record ReceiptLine(
    string Draw,
    bool Bold,
    bool Large,
    bool Center,
    bool Right,
    bool IsLeftRight,
    string LeftText,
    string RightText
);

File: spring_webapp1/src/main/java/com/example/spring_webapp1/domain/printing/ReceiptLine.java
package com.example.spring_webapp1.domain.printing;

/**
 * フォーマット済みレシートの1行を表す値オブジェクト。
 * マーカー（[B][C][R][L][LR]）解析済みの状態を保持する。
 */
public record ReceiptLine(
        String draw,
        boolean bold,
        boolean large,
        boolean center,
        boolean right,
        boolean isLeftRight,
        String leftText,
        String rightText
) {}
